### PR TITLE
Fixed problems with code generation for dynamic inspection

### PR DESCRIPTION
**Pull request by SarpHarbali**
This PR fixes issues with code generation for dynamic inspection.

## 1. Helper methods not defined in the generated code.

**Problem:**  Since the generated code only involved the method that was run with TornadoInsight, any calls to other methods caused an error.
**Solution:**  Using a recursive approach, the helper methods called within the primary method are identified and added to the generated code.
**Testing:** This fix can be tested by running dynamic inspection on BlackScholes.java, which has a call to a helper method within the class.

## 2. Class variables not defined in the generated code.

**Problem:**  Class variables were not defined in the generated code.
**Solution:**  Type, modifier, name, and value of class variables are identified and used to define them in the generated code.
**Testing:** This fix can be tested by running dynamic inspection on BFS.java, which has several class variables used by the primary method.

### CODE REVIEW

File: src/main/java/uk/ac/manchester/beehive/tornado/plugins/dynamicInspection/CodeGenerator.java [hunk 1]
```diff
@@ -31,22 +30,23 @@
...
 public class CodeGenerator {
 
-    public static void fileCreationHandler(Project project, ArrayList<PsiMethod> methods, String importCodeBlock) throws IOException {
+    public static void fileCreationHandler(Project project, ArrayList<PsiMethod> methods, ArrayList<PsiMethod> others, Map<String, Object> fields, String importCodeBlock) throws IOException {
         HashMap<String, PsiMethod> methodFile = new HashMap<>();
         File dir = FileUtilRt.createTempDirectory("files", null);
         for (PsiMethod method : methods) {
             String fileName = method.getName() + RandomStringUtils.randomAlphanumeric(5);
             File file = creatFile(project, method, importCodeBlock, fileName, dir);
+            File file = creatFile(project, method, others, importCodeBlock, fileName, dir, psiFile);
```
Comment (stratika) [inline src/main/java/uk/ac/manchester/beehive/tornado/plugins/dynamicInspection/CodeGenerator.java:42]: @SarpHarbali can you have a look? 

This line gives the following error:
```bash
error: method creatFile in class CodeGenerator cannot be applied to given types;
            File file = creatFile(project, method, importCodeBlock, fileName, dir);
                        ^
  required: Project,PsiMethod,ArrayList<PsiMethod>,Map<String,Object>,String,String,File
  found:    Project,PsiMethod,String,String,File
  reason: actual and formal argument lists differ in length
```

Comment (SarpHarbali) [reply src/main/java/uk/ac/manchester/beehive/tornado/plugins/dynamicInspection/CodeGenerator.java]: @stratika Fixed

File: src/main/java/uk/ac/manchester/beehive/tornado/plugins/service/RunInspectionAction.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void actionPerformed(@NotNull AnActionEvent e) {
             new EmptySelectionWarningDialog().show();
         }else {
             ArrayList<PsiMethod> methods = TornadoTWTask.getMethods(data);
-            DynamicInspection.process(e.getProject(), methods);
+            DynamicInspection.process(e.getProject(), methods, TornadoTWTask.getCalledMethods(methods), TornadoTWTask.getFields());
```
Comment (stratika) [inline src/main/java/uk/ac/manchester/beehive/tornado/plugins/service/RunInspectionAction.java:44]: Would it be working if you just pass the TornadoTWTask as argument and access the `getMethods()`, `getCalledMethods()` and `getFields()` methods internally?

**Review by stratika — APPROVED** [review]
LGTM

**Pull request merged**